### PR TITLE
Refactor YAP

### PR DESCRIPTION
**Pull request by gabryelreyes**
Refactor YAP for simplicity and fix unit tests

### CODE REVIEW

**Review by BlueAndi — CHANGES_REQUESTED** [review]

**Comment by BlueAndi** [comment]
Please update YAP documentation too.

File: lib/Service/YAPServer.hpp [hunk 1]
```diff
@@ -300,7 +284,7 @@ class YAPServer
             }
             else
             {
-                desync();
+                m_isSynced = false;
             }
         }
```
[the remark this answers is not in the document]
Comment (gabryelreyes) [reply lib/Service/YAPServer.hpp:289]: It would still de-sync on timeout on the next heartbeat

File: lib/Service/YAPServer.hpp [hunk 2]
```diff
@@ -80,7 +78,8 @@ class YAPServer
         m_rxAttempts(0U),
         m_numberOfTxChannels(0U),
         m_numberOfRxChannels(0U),
-        m_numberOfPendingChannels(0U)
+        m_numberOfPendingChannels(0U),
+        m_rxCallbacks{nullptr}
```
Comment (BlueAndi) [inline lib/Service/YAPServer.hpp:82]: Wrong order of initializations, please use the same order as declared. Didn't you get warnings for the target build?

**Review by BlueAndi — CHANGES_REQUESTED** [review]

**Comment by BlueAndi** [comment]
I did not review the python code ... but I got a lot of pylint warnings/errors. Please fix them.

File: lib/Service/YAPServer.hpp (hunk 2 above)
Comment (gabryelreyes) [reply lib/Service/YAPServer.hpp:82]: No warnings on my side. Do you get the warnings?

Comment (BlueAndi) [reply lib/Service/YAPServer.hpp:82]: Yes ... compiling RemoteControlTarget.

Comment (gabryelreyes) [reply lib/Service/YAPServer.hpp:82]: Should we consider adding -Werror to build flags, so the CI fails?

Comment (BlueAndi) [reply lib/Service/YAPServer.hpp]: Yes, we can do.

**Review by BlueAndi — APPROVED** [review]

**Pull request merged**